marshallers.go

```diff
@@ -0,0 +1,50 @@
+package publiccode;
+
+import (
+	"net/url"
+)
+
+type URL url.URL
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface for URLs.
+func (u *URL) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var s string
+	if err := unmarshal(&s); err != nil {
+		return err
+	}
+
+	urlp, err := url.Parse(s)
+	if err != nil {
+		return err
+	}
+	*u = (URL)(*urlp)
+
+	return nil
+}
+
+func (u *URL) MarshalYAML() (interface{}, error) {
+  return u.String(), nil
+}
+
+func (u *URL) String() string {
+	return (*url.URL)(u).String()
+}
+
+type UrlOrUrlArray []*URL
+
+func (a *UrlOrUrlArray) UnmarshalYAML(unmarshal func(interface{}) error) error {
```

Comment:
```suggestion
type UrlArray []*URL

func (a *UrlArray) UnmarshalYAML(unmarshal func(interface{}) error) error {
```

Reply:
this is actually an array

Reply:
[isBasedOn](https://docs.italia.it/italia/developers-italia/publiccodeyml-en/en/master/schema.core.html#key-isbasedon) is defined as a "string or array of strings" in the standard.

The unmarshaller normalizes it to an array of `*URL`s regardless of the original input: in case of a string it gets normalized as an array with a single element.

I have an entry in my notes to discuss this very issue: I think it would be nice to have `isBasedOn` defined as just an array of string.